pkg/resources/pullSecret.go

```diff
@@ -54,3 +54,43 @@ func CopySecret(ctx context.Context, client k8sclient.Client, srcName, srcNamesp
 
 	return err
 }
+
+func LinkSecretToServiceAccounts(ctx context.Context, client k8sclient.Client, namespace string, secretName string) error {
+	serviceAccounts := &corev1.ServiceAccountList{}
+	listOpts := []k8sclient.ListOption{
+		k8sclient.InNamespace(namespace),
+	}
+	err := client.List(ctx, serviceAccounts, listOpts...)
+	if err != nil {
+		return err
+	}
+
+	for _, sa := range serviceAccounts.Items {
+		currentSa := &corev1.ServiceAccount{}
+		err = client.Get(ctx, types.NamespacedName{Namespace: namespace, Name: sa.Name}, currentSa)
+		if err != nil {
+			return err
+		}
+
+		pullSecretFound := false
+		for _, ips := range currentSa.ImagePullSecrets {
+			if ips.Name == secretName {
+				pullSecretFound = true
```

Comment:
I think we can stop the loop once the pull secret is found

Reply:
Good point, i'll change that